### PR TITLE
[Core] Adding the  initialize and finalize to the constraints

### PR DESCRIPTION
**Pull request by loumalouomega**
This adds the missing initialize and finalize for constraints to the schemes

This should be migrated to the Release7.0 branch

### CODE REVIEW

File: kratos/solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h [hunk 1]
```diff
@@ -192,20 +192,33 @@ class ResidualBasedIncrementalUpdateStaticScheme
...
-        // Initialize solution step for all of the conditions
+        // Definition of the first condition iterator
+        const auto it_cond_begin = rModelPart.ConditionsBegin();
+
         #pragma omp parallel for
-        for(int i=0; i<static_cast<int>(rModelPart.Conditions().size()); i++) {
-            auto it_cond = rModelPart.ConditionsBegin() + i;
+        for(int i=0; i<static_cast<int>(rModelPart.Conditions().size()); ++i) {
+            auto it_cond = it_cond_begin + i;
             it_cond->InitializeNonLinearIteration(r_current_process_info);
```
Comment (philbucher) [inline kratos/solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h:210]: why not
```suggestion
            (it_cond_begin + i)->InitializeNonLinearIteration(r_current_process_info);
```
?

Comment (loumalouomega) [reply kratos/solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h:210]: Well, we can do that, but it is not so relevant (and taking into account that for the Release 7.0 we must cherrypick the less commits the better)

**Comment by adityaghantasala** [comment]
Looks good. Out of curiosity , are you using the those functions from constraints in any computations ?

**Comment by loumalouomega** [comment]
Not yet, but will be soon

**Review by adityaghantasala — APPROVED** [review]
Good for me !

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**